### PR TITLE
feat: upstream aws-sso

### PR DESCRIPTION
**Pull request by Gowiem**
## what
* Upstreams the aws-sso component

## why
* Keeping our components up-to-date with updates that have been made for clients

### CODE REVIEW

File: modules/aws-sso/README.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      - ReadOnlyAccess
+
+components:
+  terraform:
+    aws-sso:
+      vars:
+        account_assignments:
+          audit:
+            groups:
+              Production Cloud Infrastructure Engineers:
```
Comment (osterman) [inline modules/aws-sso/README.md:27]:
```suggestion
              Some Group in IDP:
```

File: modules/aws-sso/README.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                permission_sets:
+                  - AdministratorAccess
+                  - ReadOnlyAccess
+              Production Cloud Engineers:
+                permission_sets:
+                  - ReadOnlyAccess
+          auto:
+            groups:
+              <<: *prod-cloud-engineers
+              Production Cloud Engineers:
```
Comment (osterman) [inline modules/aws-sso/README.md:37]:
```suggestion
              A Less Special Group in IDP:
```

File: modules/aws-sso/README.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+## Usage
+
+**Stack Level**: Global
+
+The example snippet below shows how to use this module with various combinations (plain YAML, YAML Anchors and a combination of the two):
+
+```yaml
+prod-cloud-engineers: &prod-cloud-engineers
+  Production Cloud Infrastructure Engineers:
```
Comment (osterman) [inline modules/aws-sso/README.md:15]:
```suggestion
  Production Cloud Admins:
```

File: modules/aws-sso/README.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+This component assumes that AWS SSO has already been enabled via the AWS Console (there isn't terraform or AWS CLI support for this currently) and that the IdP has been configured to sync users and groups to AWS SSO.
+
+## Usage
+
+**Stack Level**: Global
+
+The example snippet below shows how to use this module with various combinations (plain YAML, YAML Anchors and a combination of the two):
+
+```yaml
+prod-cloud-engineers: &prod-cloud-engineers
```
Comment (osterman) [inline modules/aws-sso/README.md:14]:
```suggestion
prod-cloud-admins: &prod-cloud-admins
```

File: modules/aws-sso/README.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+              Production Cloud Infrastructure Engineers:
+                permission_sets:
+                  - AdministratorAccess
+                  - ReadOnlyAccess
+              Production Cloud Engineers:
+                permission_sets:
+                  - ReadOnlyAccess
+          auto:
+            groups:
+              <<: *prod-cloud-engineers
```
Comment (osterman) [inline modules/aws-sso/README.md:36]:
```suggestion
              <<: *prod-cloud-devs
```

File: modules/aws-sso/README.md (hunk 2 above)
Comment (osterman) [inline modules/aws-sso/README.md:37]:
```suggestion
              "Production Cloud Devs":
```

**Comment by Gowiem** [comment]
Old as dirt!

**Pull request closed without merging**